src/resmom/start_exec.c

```diff
@@ -2179,22 +2179,22 @@ receive_string_data(int downfds, int  upfds)
 	/* ack that we got the r_size */
 	ack_size = sizeof(size_t);
 	if (write_pipe_data(upfds, &ack_size, sizeof(size_t)) != 0) {
-		snprintf(log_buffer, sizeof(log_buffer), "write of %lu to pipe failed", ack_size);
+		snprintf(log_buffer, sizeof(log_buffer), "write of %lu to pipe failed", (unsigned long)ack_size);
 		log_err(errno, __func__, log_buffer);
 		return (NULL);
 	}
 
 	/* now get the actual string data */
 	r_buf = read_pipe_data(downfds, r_size, PIPE_READ_TIMEOUT);
 	if (r_buf == NULL) {
-		snprintf(log_buffer, sizeof(log_buffer), "read of pipe of size %lu for failed", r_size);
+		snprintf(log_buffer, sizeof(log_buffer), "read of pipe of size %lu for failed", (unsigned long)r_size);
 		log_err(errno, __func__, log_buffer);
 		return (NULL);
 	}
 	/* send back as an acknowledgement that MOM got it */
 	ack_size = r_size;
 	if (write_pipe_data(upfds, &ack_size, sizeof(size_t)) != 0) {
-		snprintf(log_buffer, sizeof(log_buffer), "write of %lu to pipe failed", r_size);
+		snprintf(log_buffer, sizeof(log_buffer), "write of %lu to pipe failed", (unsigned long)r_size);
```

Comment:
You might want to add "bytes" after "%lu" in each of these messages to convey the fact they are a measure of size. Something like "write of length 1234 bytes to pipe failed"

Reply:
Yes. Message updated.